src/software/ai/hl/stp/play/play.cpp

```diff
@@ -310,6 +310,23 @@ Play::assignTactics(const WorldPtr &world_ptr, TacticVector tactic_vector,
             double robot_cost_for_tactic =
                 primitives.at(robot.id())->getEstimatedPrimitiveCost();
 
+            std::set<int> prioritized_robot_ids = tactic->prioritizedRobotIds();
+            if (prioritized_robot_ids.contains(robot.id()))
+            {
+                // tune the cost of the prioritized robot so it gets selected more for this tactic
+                robot_cost_for_tactic *= PRIORITIZED_ROBOT_COST_TUNE;
+            }
+
+            if (robot.hasLastStatusTime())
+            {
+                auto time_since_last_status_s = world_ptr->getMostRecentTimestamp() - robot.getLastStatusTime();
+                if (time_since_last_status_s > Duration::fromMilliseconds(DISCONNECT_DURATION_MS))
+                {
+                    // increase the cost arbitrarily to exclude non-responsive robots
+                    robot_cost_for_tactic *= 10;
+                }
+            }
+
```

Comment:
Can you explain the intention to me?

To me, it seems that the goal of this PR is to stop robots that don't seem to work to be assigned important tactics.

This change multiplies the "cost" for every single tactic that a working robot could take on. In effect, I don't think that would change whether a "prioritized" robot gets assigned a more important tactic or not because all the tactics that this robot could take on have inflated costs.

Let me know if I'm misunderstanding this